src/components/PlayerSubmissionForm.js

```diff
@@ -3,24 +3,54 @@ import PropTypes from 'prop-types';
 
 import './PlayerSubmissionForm.css';
 
-const PlayerSubmissionForm = () => {
-  return (
-    <div className="PlayerSubmissionForm">
-      <h3>Player Submission Form for Player #{  }</h3>
+const PlayerSubmissionForm = props => {
+  const defaultFormFields = {
+    adj1: '',
+    noun1: '',
+    adv: '',
+    verb: '',
+    adj2: '',
+    noun2: ''
+  };
 
-      <form className="PlayerSubmissionForm__form" >
+  const [formFields, setFormFields] = useState(defaultFormFields);
 
-        <div className="PlayerSubmissionForm__poem-inputs">
+  const onInputChange = event => {
+    const newFormFields = { ...formFields };
+    newFormFields[event.target.name] = event.target.value;
+    setFormFields(newFormFields);
+  };
 
-          {
-            // Put your form inputs here... We've put in one below as an example
-          }
-          <input
-            placeholder="hm..."
-            type="text" />
+  const onFormSubmit = event => {
+    event.preventDefault();
+    props.sendSubmission(formFields);
+    setFormFields(defaultFormFields);
+  };
 
+  return (
+    <div className="PlayerSubmissionForm">
+      <h3>Player Submission Form for Player #{ props.index }</h3>
+      <form className="PlayerSubmissionForm__form" onSubmit={ onFormSubmit } >
+        <div className="PlayerSubmissionForm__poem-inputs">
+          { props.fields.map((value, index) => {
+            if (typeof value === 'string') {
+              return value;
+            } else {
+              return <input 
+                        key={ index }
+                        name={ value.key }
+                        placeholder={ value.placeholder }
+                        onChange={ onInputChange }
+                        type='text' 
+                        value={ formFields[value.key] || '' }
+                        className={ !formFields[value.key]
+                                    ? 'PlayerSubmissionForm__input--invalid' 
+                                    : '' 
+                                  }
+                    />
```

Comment:
I think in this mix of things, React is expecting each element of the returned array to have a key value, which is why you're getting the warning.

Reply:
Ah thank you!  Looks like I needed to add keys to the two items in my array with the components:

[<RecentSubmission key={ poemLines.length } submission={ poemLines[poemLines.length - 1] } />,
<PlayerSubmissionForm key={ playerNum } index={ playerNum } sendSubmission={ addLine } fields={ FIELDS } />]